drivers/spi/Kconfig

```diff
@@ -99,250 +53,185 @@ config SYS_LOG_SPI_LEVEL
 
 config SPI_CS_GPIO
 	bool "SPI port CS pin is controlled via a GPIO port"
-	depends on GPIO
+	depends on GPIO && SPI_LEGACY_API
```

Comment:
Shouldn't it be `!SPI_LEGACY_API`?

Reply:
no it's ok here, Kconfig based CS_GPIO was meant for legacy drivers. 

Now it's fully externalized in spi api, so there is no need of a Kconfig option

Reply:
Right. I misunderstood this. It's ok.